pkg/egressgateway/manager_privileged_test.go

```diff
@@ -159,84 +157,6 @@ func (k *EgressGatewayTestSuite) TestEgressGatewayManager(c *C) {
 	ep1, id1 := newEndpointAndIdentity("ep-1", ep1IP, ep1Labels)
 	egressGatewayManager.OnUpdateEndpoint(&ep1)
 
-	assertEgressRules(c, []egressRule{
-		{ep1IP, destCIDR, egressIP1, egressIP1},
-	})
-	assertIPRules(c, []ipRule{})
-
-	// Update the labels for ep1 in order for it to NOT be a match
-	id1 = updateEndpointAndIdentity(&ep1, id1, map[string]string{})
-	egressGatewayManager.OnUpdateEndpoint(&ep1)
-
-	assertEgressRules(c, []egressRule{})
-	assertIPRules(c, []ipRule{})
-
-	// Restore the old ep1 lables in order for it to be a match
-	id1 = updateEndpointAndIdentity(&ep1, id1, ep1Labels)
-	egressGatewayManager.OnUpdateEndpoint(&ep1)
-
-	assertEgressRules(c, []egressRule{
-		{ep1IP, destCIDR, egressIP1, egressIP1},
-	})
-	assertIPRules(c, []ipRule{})
-
-	// Create a new policy
-	policy2 := newEgressPolicyConfig("policy-2", ep2Labels, destCIDR, egressIP2)
-	egressGatewayManager.OnAddEgressPolicy(policy2)
-
-	assertEgressRules(c, []egressRule{
-		{ep1IP, destCIDR, egressIP1, egressIP1},
-	})
-	assertIPRules(c, []ipRule{})
-
-	// Add a new endpoint and ID which matches policy-2
-	ep2, _ := newEndpointAndIdentity("ep-2", ep2IP, ep2Labels)
-	egressGatewayManager.OnUpdateEndpoint(&ep2)
-
-	assertEgressRules(c, []egressRule{
-		{ep1IP, destCIDR, egressIP1, egressIP1},
-		{ep2IP, destCIDR, egressIP2, egressIP2},
-	})
-
-	// Update the endpoint labels for policy-1 in order for it to NOT be a match
-	updateEndpointAndIdentity(&ep1, id1, map[string]string{})
-	egressGatewayManager.OnUpdateEndpoint(&ep1)
-
-	assertEgressRules(c, []egressRule{
-		{ep2IP, destCIDR, egressIP2, egressIP2},
-	})
-	assertIPRules(c, []ipRule{})
-
-	// Restore the old endpoint lables in order for it to be a match
-	id1 = updateEndpointAndIdentity(&ep1, id1, ep1Labels)
-	egressGatewayManager.OnUpdateEndpoint(&ep1)
-
-	assertEgressRules(c, []egressRule{
-		{ep1IP, destCIDR, egressIP1, egressIP1},
-		{ep2IP, destCIDR, egressIP2, egressIP2},
-	})
-
-	// Delete policy-1 and policy-2 policies
-	egressGatewayManager.OnDeleteEgressPolicy(policy1.id)
-	egressGatewayManager.OnDeleteEgressPolicy(policy2.id)
-
-	// As well as ep-1 and ep-2 endpoints
-	egressGatewayManager.OnDeleteEndpoint(&ep1)
-	egressGatewayManager.OnDeleteEndpoint(&ep2)
-
-	assertEgressRules(c, []egressRule{})
-	assertIPRules(c, []ipRule{})
-
-	// Create a new policy
-	policy3 := newEgressPolicyConfigWithNodeSelector("policy-3", ep1Labels, destCIDR, nodeGroup1Selector, testInterface1)
-	egressGatewayManager.OnAddEgressPolicy(policy3)
-
-	assertEgressRules(c, []egressRule{})
-	assertIPRules(c, []ipRule{})
-
-	// Add back ep-1 endpoint which matches policy-3
-	egressGatewayManager.OnUpdateEndpoint(&ep1)
-
```

Comment:
I didn't quite catch, but what about the change in CRD format means that this test code is no longer necessary?

Reply:
CEGP mandates a NodeSelector to pick the Gateway node (while CENP implicitly picked the node based on the configured `EgressIP`). So with CENP gone, we don't need to/can't test a policy without NodeSelector.